src/instrument/mongoose_instrument_exometer.erl

```diff
@@ -160,15 +164,27 @@ update_metric(Name, histogram, Value) when is_integer(Value) ->
 -spec exometer_metric_name(mongoose_instrument:event_name(), mongoose_instrument:labels(),
                            mongoose_instrument:metric_name()) -> exometer:name().
 exometer_metric_name(EventName, Labels, MetricName) ->
-    [get_host_type_prefix(Labels), EventName] ++ exometer_labels(Labels) ++ [MetricName].
+    HostPrefix = get_host_type_prefix(Labels),
+    RemainingLabels = maps:without([host_type], Labels),
+    ExometerLabels = exometer_labels(RemainingLabels),
+    [HostPrefix, EventName] ++ ExometerLabels  ++ [MetricName].
 
 %% This logic will need extending if we add more labels
-exometer_labels(#{function := Function}) ->
+exometer_labels(#{function := Function} = Labels) when map_size(Labels) =:= 1 ->
     [Function];
-exometer_labels(#{pool_tag := PoolTag}) ->
+exometer_labels(#{pool_tag := PoolTag} = Labels) when map_size(Labels) =:= 1 ->
     [PoolTag];
-exometer_labels(#{}) ->
-    [].
+exometer_labels(#{pool_id := PoolId} = Labels) when map_size(Labels) =:= 1 ->
+    [PoolId];
+exometer_labels(#{connection_type := ConnectionType} = Labels) when map_size(Labels) =:= 1 ->
```

Comment:
This is not included in the type `mongoose_instrument:labels()`, so this label should never occur. Unless someone forgot to update that type.